site/app/views/calendar/CalendarView.php

```diff
@@ -40,11 +40,25 @@ public function showCalendar(CalendarInfo $info, array $courses, bool $in_course
         //Create list of courses and their term and get color
         $formatted_courses = [];
         $unformatted_courses = [];
+        $display_names = [];
         //$course_colors = [];
         foreach ($courses as $course) {
             $course_string = sprintf("%s %s", $course->getTitle(), $course->getTerm());
             array_push($formatted_courses, $course_string);
         }
+        foreach ($courses as $course) {
+            if ($course->getDisplayName()) {
+                $name_string = sprintf("%s %s", $course->getDisplayName(), $course->getTerm());
+            }
+            else {
+                $name_string = sprintf("%s %s", $course->getTitle(), $course->getTerm());
+            }
+            array_push($display_names, $name_string);
+        }
+        $combinedArray = [];
+        foreach (array_map(null, $formatted_courses, $display_names) as [$course, $name]) {
+            $combinedArray[] = ['course' => $course, 'name' => $name];
+        }
```

Comment:
```suggestion
$combinedArray = [];

foreach ($courses as $course) {
    $displayName = $course->getDisplayName();
    $title = $course->getTitle();
    $term = $course->getTerm();
    
    $name_string = ($displayName ? $displayName : $title) . ' ' . $term;
    
    $combinedArray[] = ['course' => $course, 'name' => $name_string];
}

```

Reply:
I refactored this part of the code. In the Combinearray( renamed to courseWithName) we need to use the formatted_courses instead of course.